api/peripheral_api/util/RFID-helpers.js

```diff
@@ -0,0 +1,195 @@
+  require('../../util/constants').STATUS_CODES;
+const fs = require('fs');
+
+
+/**
+ * Abstracts RFID card backend logic to a class to improve readability.
+ * @member {bool} addRfid - Keeps track of the state of the server, whether
+ * card data from the ESP32 should be verified or added to the database.
+ * @member {string|null} name - The name of the user we should add to
+ * MongoDB along with card data. When the server is in the add RFID state,
+ * this variable is populated with said name.
+ */
+class RfidHelper {
+  constructor() {
+    this.addRfid = false;
+    this.name = null;
+    this.inTesting = false;
+  }
+
+  /**
+   * Helper function to determine the state of the RFID API.
+   * @returns {bool} If we are adding a card the value is true,
+   * else false for verifying.
+   */
+  addingRfid() {
+    return this.addRfid;
+  }
+
+  /**
+   * Determines if the required files are present for the RFID server to work.
+   * We check if an array of file names are present in api/config/AWS-IOT/
+   * directory. A boolean value is returned depending on their presence.
+   * @returns {bool} If the files are present the value is true,
+   * else false.
+   */
+  keysExist() {
+    const filePaths = ['private.pem.key', 'cert.pem.crt', 'AmazonRootCA1.pem'];
+    for (const filePath of filePaths) {
+      try {
+        if (!fs.existsSync('../api/config/AWS-IOT/' + filePath)) {
+          return false;
+        }
+      } catch (e) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Helper function to determine if currently running in testing
+   * @return If running api tests then it return true else returns false
+   */
+  testing() {
+    return this.inTesting;
+  }
+
+  /**
+   * Helper function to hash the RFID data for security purposes
+   * @param {String|null} byte RFID card data
+   * @returns a hashed value if successful else null
+   */
+  hashedCardData(byte) {
+    return new Promise((resolve, reject) => {
+      bcrypt.genSalt(10, function(error, salt) {
+        if (error) {
+          resolve(null);
+        }
+        bcrypt.hash(byte, secretKey, function(hashError, hash) {
+          if (hashError) {
+            resolve(null);
+          }
+          resolve(hash);
+        });
+      });
+    });
+  }
+
+  /**
+   * Retrieves all RFID cards in the database
+   * @returns {Promise<Array<Object>>}
+   */
+  getCards() {
+    return new Promise((resolve, reject) => {
+      RFID.find()
+        .then((items) => resolve(items))
+        .catch((error) => {
+          resolve([]);
+        });
+    });
+  }
+
+  /**
+   * Validates the scanned RFID card by comparing the hashed data
+   * with the one in the database
+   * @param {JSONObject} payload Incoming card data from ESP32
+   * @returns {Number} 404 if card data was null or not in the database
+   * else 200 if the incoming data is not null and it exists in the database
+   */
+  async validate(payload) {
```

Comment:
little more descriptive
```suggestion
  async validateCard(payload) {
```